dhcp4relay/src/dhcp4relay.cpp

```diff
@@ -0,0 +1,1120 @@
+#include <errno.h>
+#include <event.h>
+#include <event2/bufferevent.h>
+#include <event2/event.h>
+#include <fcntl.h>
+#include <pcapplusplus/DhcpLayer.h>
+#include <pcapplusplus/EthLayer.h>
+#include <pcapplusplus/IPv4Layer.h>
+#include <pcapplusplus/Packet.h>
+#include <pcapplusplus/PayloadLayer.h>
+#include <pcapplusplus/UdpLayer.h>
+#include <signal.h>
+#include <unistd.h>
+
+#include <sstream>
+
+#include "configdb.h"
+#include "dhcp4_sender.h"
+#include "dhcp4relay_mgr.h"
+#include "dhcp4relay_stats.h"
+#include "sonicv2connector.h"
+
+struct event_base *base;
+struct event *ev_sigint;
+struct event *ev_sigterm;
+static std::string vlan_member = "VLAN_MEMBER|";
+static std::string counter_table = "DHCPv4_COUNTER_TABLE|";
+
+extern std::string host_mac_addr;
+extern std::string hostname;
+
+static uint8_t client_recv_buffer[BUFFER_SIZE];
+int config_pipe[2];
+
+/* DHCPv4 filter */
+/* sudo tcpdump -dd "inbound and ip dst 255.255.255.255 && udp dst port 67" */
```

Comment:
Is this sock_filter the RX_socket in below diagram?
![image](https://github.com/user-attachments/assets/b4aefdcf-d36d-4af4-bbc4-bbb2fbf0e0e4)

Reply:
Yes, it's for the RX_socket. In the current implementation, we were assuming that client always requires a broadcast response. But you have a valid point, we should suport both unicast and broadcast response from Server to Client. We will fix it.